homeassistant/components/hydrawise.py

```diff
@@ -0,0 +1,173 @@
+"""
+Support for Hydrawise cloud.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/hydrawise
+"""
+import asyncio
+from datetime import timedelta
+import logging
+
+from requests.exceptions import ConnectTimeout, HTTPError
+import voluptuous as vol
+
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, CONF_ACCESS_TOKEN, CONF_SCAN_INTERVAL)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect, dispatcher_send)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_interval
+
+REQUIREMENTS = ['hydrawiser==0.1.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+ALLOWED_WATERING_TIME = [5, 10, 15, 30, 45, 60]
+
+CONF_ATTRIBUTION = "Data provided by hydrawise.com"
+CONF_WATERING_TIME = 'watering_minutes'
+
+NOTIFICATION_ID = 'hydrawise_notification'
+NOTIFICATION_TITLE = 'Hydrawise Setup'
+
+DATA_HYDRAWISE = 'hydrawise'
+DOMAIN = 'hydrawise'
+DEFAULT_WATERING_TIME = 15
+
+KEY_MAP = {
+    'auto_watering': 'Automatic Watering',
+    'is_watering': 'Watering',
+    'manual_watering': 'Manual Watering',
+    'next_cycle': 'Next Cycle',
+    'status': 'Status',
+    'watering_time': 'Watering Time',
+    'rain_sensor': 'Rain Sensor',
+}
+
+ICON_MAP = {
+    'auto_watering': 'mdi:autorenew',
+    'is_watering': '',
+    'manual_watering': 'mdi:water-pump',
+    'next_cycle': 'mdi:calendar-clock',
+    'status': '',
+    'watering_time': 'mdi:water-pump',
+    'rain_sensor': ''
+}
+
+UNIT_OF_MEASUREMENT_MAP = {
+    'auto_watering': '',
+    'is_watering': '',
+    'manual_watering': '',
+    'next_cycle': '',
+    'status': '',
+    'watering_time': 'min',
+}
+
+BINARY_SENSORS = ['is_watering', 'status', 'rain_sensor']
+
+SENSORS = ['next_cycle', 'watering_time']
+
+SWITCHES = ['auto_watering', 'manual_watering']
+
+SCAN_INTERVAL = timedelta(seconds=20)
```

Comment:
Is there a benefit to go with 20 s instead of the default of 30 s?

Reply:
No benefit. I changed back to 30.